src/test/java/com/iexec/core/security/EIP712ChallengeServiceTests.java

```diff
@@ -43,6 +44,24 @@ void shouldGetSameChallengeForSameWallet() {
         assertThat(challenge1).usingRecursiveComparison().isEqualTo(challenge2);
     }
 
+    @Test
+    void shouldGetSameChallengeWithChecksumWallet() {
+        String checksumWallet = Keys.toChecksumAddress(WALLET_ADDRESS_1);
+        assertThat(checksumWallet).isNotEqualTo(WALLET_ADDRESS_1);
+        assertThat(challengeService.getChallenge(checksumWallet))
+                .usingRecursiveComparison()
+                .isEqualTo(challengeService.getChallenge(WALLET_ADDRESS_1));
+    }
+
+    @Test
+    void shouldNotGetSameChallengeWithUppercaseWallet() {
```

Comment:
Actually, from an Ethereum blockchain perspective, addresses are the same whether they are `lowercase`, `uppercase`, or `checksum`.